A1-2015/SystemC/MergeFilter.java

```diff
@@ -32,40 +34,48 @@ public void run() {
       Frame newestMergeInputFrame = new Frame();
       
       while(true) {
-          if(this.InputFilter.isAlive() && this.mergeInputFilter.isAlive()){
-            if(inputStreamActive){
-                newestInputFrame = ReadFrameInput(this.InputReadPort);
-                System.out.print("Read frame from input stream: " + newestInputFrame.toString() + "\n");
-                if(newestInputFrame.timestamp.getTime() <= newestMergeInputFrame.timestamp.getTime()){
-                    WriteFrameOutput(this.OutputWritePort, newestInputFrame);
-                } else {
-                    inputStreamActive = false;
-                    WriteFrameOutput(this.OutputWritePort, newestMergeInputFrame);
-                }
-            }else{
-                newestMergeInputFrame = ReadFrameInput(this.mergeInputReadPort);
-                System.out.print("Read frame from merge stream: " + newestMergeInputFrame.toString() + "\n");
-                if(newestMergeInputFrame.timestamp.getTime() <= newestInputFrame.timestamp.getTime()){
-                    WriteFrameOutput(this.OutputWritePort, newestMergeInputFrame);
-                } else {
-                    inputStreamActive = true;
-                    WriteFrameOutput(this.OutputWritePort, newestInputFrame);
-                }
-                
-            }
-          }else {
-              if(this.InputFilter.isAlive()){
-                  while(this.InputFilter.isAlive()){
-                      newestInputFrame = ReadFrameInput(this.InputReadPort);
-                      WriteFrameOutput(this.OutputWritePort, newestInputFrame);
+          if(inputStreamActive){

```

Comment:
does this boolean consider all the inputs?

Reply:
I should probably rename it, it refers to which of the two streams the algorithm is currently reading from, it switches back and forth